kubernetes/client_factory.go

```diff
@@ -493,18 +440,22 @@ func (cf *clientFactory) getConfig(clusterInfo *RemoteClusterInfo) (*rest.Config
 			return nil, err
 		}
 
+		// Set the token file so the underlying client can refresh it when needed.
+		remoteConfig.BearerTokenFile = clusterInfo.SecretFile
```

Comment:
I'm still leery of manually overriding these settings and I think this already gets set by `client-go` if you have a `SecretFile` in your kubeconfig?

Reply:
not set. I change this line 444 to this:
```go
		//MAZZ DELETE remoteConfig.BearerTokenFile = clusterInfo.SecretFile
		log.Errorf("MAZZ2 btf=[%v];sf=[%v]", remoteConfig.BearerTokenFile, clusterInfo.SecretFile)
```
and in the logs I see:
```
2024-01-17T19:09:02Z ERR MAZZ2 btf=[];sf=[/kiali-remote-cluster-secrets/kiali-remote-cluster-secret-west/west]
```

Reply:
But yet when I leave line 444 the way it is here, I get this error:
```
2024-01-17T19:12:17Z ERR MAZZ2 btf=[/kiali-remote-cluster-secrets/kiali-remote-cluster-secret-west/west];sf=[/kiali-remote-cluster-secrets/kiali-remote-cluster-secret-west/west]
...
W0117 19:12:27.423526       1 reflector.go:535] pkg/mod/k8s.io/client-go@v0.28.3/tools/cache/reflector.go:229: failed to list *v1.ReplicaSet: Get "https://172.18.0.5:6443/apis/apps/v1/replicasets?limit=500&resourceVersion=0": net/http: invalid header field value for "Authorization"
```
I suspect I don't understand what value the BearerTokenFile should be.

Reply:
I was assuming "BearerTokenFile" is the same as the kubeconfig file, but apparently it is not. The BearerTokenFile can be specified in the kubeconfig, and I believe that is what should get set here. We do not have BearerTokenFiles defined in our generated remote cluster secrets.

So I have a feeling this client-go feature is for something slightly different that what we are expecting to be able to use it. I don't think this is something we can use (i.e. I think we should keep our code that refreshes).

@nrfox See if you understand this differently than me, and let me know what I'm missing.

Reply:
> I was assuming "BearerTokenFile" is the same as the kubeconfig file, but apparently it is not.

`BearerTokenFile` is not the same as the kubeconfig. It is a file that contains the `BearerToken`. In the case of the `InClusterConfig` it is just `/var/run/secrets/kubernetes.io/serviceaccount/token`.

Reply:
OK, I think I get it. I was thinking this was going to refresh everything, not just the in-cluster client.

Reply:
I removed this.